src/discljord/cdn.clj

```diff
@@ -0,0 +1,72 @@
+(ns discljord.cdn
+  "Namespace with functions to create [cdn urls](https://discord.com/developers/docs/reference#image-formatting) to image data from API entities such as users or guilds."
+  (:require [clojure.string :refer [starts-with?]]))
+
+(def base-url "https://cdn.discordapp.com")
+
+(defn custom-emoji
+  "Takes a custom emoji object or id and returns a url to the image data.
+  Note that an id alone is not enough to determine whether the emoji is animated or not,
+  so if you pass an id you will always get a url to a png."
+  [emoji]
+  (str base-url "/emojis/"
+       (cond-> emoji (map? emoji) :id) \. (if (:animated emoji) "gif" "png")))
+
+(defn animated?
+  "Takes an image data hash and returns whether it represents an animated image (gif) or not."
+  [hash]
+  (starts-with? hash "a_"))
+
+(defn file-name
+  "Takes an image data hash and returns the file name it represents with the correct extension."
+  [hash]
+  (str hash \. (if (animated? hash) "gif" "png")))
+
+(defn image-url-generator
+  "Returns a function that, given an object with an id and an image data hash, returns a url to that image."
+  [path hash-key]
+  (fn [{:keys [id] hash hash-key}]
```

Comment:
Are you intending to bind the value under the symbol `hash-key` to the symbol `hash`? I think this destructuring is a bit wrong, you'll probably need to do something more like so:
```clojure
(fn [{id :id :as entity}]
  (let [hash (entity hash-key)]
    ;; ... more stuff ...
    ))
```

Reply:
No, that destructuring is fine. `hash-key` will be resolved to its value when destructuring, i.e. it will destructure `:avatar` for users, `:icon` for guilds and so on. I have tested it, but I can change it for clarity if you like.

Reply:
That's news to me, I didn't think that it evaluated the item on the right. Well if that works, I see no reason to change it.